Gemfile

```diff
@@ -3,6 +3,10 @@ source "http://rubygems.org"
 gem 'json', '~> 1.8', '>= 1.8.6'
 gem 'algoliasearch', '>= 1.17.0', '< 2.0.0'
 
+if defined?(RUBY_VERSION) && RUBY_VERSION == "1.8.7"
+  gem 'ruby18_source_location'
+end
+
```

Comment:
Why do we need that @julienbourdeau ? Doesn't seem to be a very popular library.

Reply:
`source_location` is only available after in ruby 1.9, this gem backports the feature to 1.8.
I thought `__file__` should work with ruby < 1.9 but I got undefined method in Travis as well.

One other solution would be to not have this feature in Ruby < 1.9 and in this case we always call `attr_changed?` method. In this case, people won't be able to use the new name but you can't run Rails 5 on ruby 1.8, I don't think it's important.

Now, that I think about it, It might be the best solution. WDYT?

Reply:
I think always calling the `attr_changed?` method is fine, it's what we've done before; no?

> In this case, people won't be able to use the new name but you can't run Rails 5 on ruby 1.8, I don't think it's important.

Correct

> Now, that I think about it, It might be the best solution. WDYT?

Most probably. Also, what you've done is not enough: the `Gemfile` is handling the dependencies of the project, while the dependencies that are gonna be bundled while releasing the gem is defined in `algoliasearch-rails.gemspec`. There should be a single mutualized list for both, but the fact we're having extra dependencies based on the ruby VM/versions makes it impossible.

Reply:
Thank you Sylvain for reminding me about `algoliasearch-rails.gemspec`. When writing this "fix" it totally went over my head 😅 🙏